examples/moonshot-mainseries/01-simulate-receptors.py

```diff
@@ -0,0 +1,228 @@
+log.info(f':chains:  Adding virtual bonds between molecules')
+custom_bond_force = openmm.CustomBondForce('0')
+for molecule_index in range(len(molecules_atom_indices)-1):
+    custom_bond_force.addBond(molecules_atom_indices[molecule_index][0], molecules_atom_indices[molecule_index+1][0], [])
+system.addForce(custom_bond_force)
+
+# Add barostat
+from openmm import MonteCarloBarostat
+system.addForce(MonteCarloBarostat(pressure, temperature));
+log.info(f':game_die: Default Periodic box:')
+for dim in range(3):
+    log.info(f'  :small_blue_diamond: {system.getDefaultPeriodicBoxVectors()[dim]}')
+
+# Create integrator
+log.info(':building_construction:  Creating integrator...')
+from openmm import LangevinMiddleIntegrator
+integrator = LangevinMiddleIntegrator(temperature, collision_rate, timestep)
+
+# Create simulation
+log.info(':mage:  Creating simulation...')
+from openmm.app import Simulation
+simulation = Simulation(modeller.topology, system, integrator, platform=platform)
+context = simulation.context
+context.setPositions(modeller.positions)
+
+# Write initial PDB, if requested.
+if arguments['--initial']:
+    log.info(f":page_facing_up:  Writing initial PDB to {arguments['--initial']}")
+    output_positions = context.getState(getPositions=True, enforcePeriodicBox=False).getPositions(asNumpy=True)
+    with open(arguments['--initial'], 'w') as outfile:    
+        PDBFile.writeFile(output_topology, output_positions[output_indices,:], file=outfile, keepIds=False);
+
+# Minimize energy
+log.info(':skier:  Minimizing ...')
+simulation.minimizeEnergy();
+
+# Write minimized PDB, if requested
+if arguments['--minimized']:
+    log.info(f":page_facing_up:  Writing minimized PDB to {arguments['--minimized']}")
+    output_positions = context.getState(getPositions=True, enforcePeriodicBox=False).getPositions(asNumpy=True)
+    with open(arguments['--minimized'], 'w') as outfile:    
+        PDBFile.writeFile(output_topology, output_positions[output_indices,:], file=outfile, keepIds=False);
+
+# Equilibrate
+log.info(':fire:  Heating ...')
+simulation.context.setVelocitiesToTemperature(temperature);
+simulation.step(equilibration_steps);
+
+# Add reporter to generate DCD trajectory, if requested
+if arguments['--dcdtraj']:
+    log.info(f":page_facing_up:  Will write DCD trajectory to {arguments['--dcdtraj']}")
+    from mdtraj.reporters import DCDReporter
+    simulation.reporters.append(DCDReporter(arguments['--dcdtraj'], reporting_interval, atomSubset=output_indices))
+
+# Add reporter to generate XTC trajectory, if requested
+if arguments['--xtctraj']:
+    log.info(f":page_facing_up:  Will write XTC trajectory to {arguments['--xtctraj']}")
+    from mdtraj.reporters import XTCReporter
+    simulation.reporters.append(XTCReporter(arguments['--xtctraj'], reporting_interval, atomSubset=output_indices))
+
+# Add reporter to generate PDB trajectory, if requested
+# NOTE: The PDBReporter does not currently support atom subsets
```

Comment:
is there any way to do this in the future? For ligand<->residue contacts just writing out the ligand + selected residues would be super useful..

Reply:
Yes, it should be easy--I've opened an issue: https://github.com/openmm/openmm/issues/3925